crates/noirc_evaluator/src/ssa_refactor/ir/instruction.rs

```diff
@@ -338,6 +314,110 @@ impl Instruction {
     }
 }
 
+/// Try to simplify this cast instruction. If the instruction can be simplified to a known value,
+/// that value is returned. Otherwise None is returned.
+fn simplify_cast(value: ValueId, dst_typ: &Type, dfg: &mut DataFlowGraph) -> SimplifyResult {
+    use SimplifyResult::*;
+    if let Some(constant) = dfg.get_numeric_constant(value) {
+        let src_typ = dfg.type_of_value(value);
+        match (src_typ, dst_typ) {
+            (Type::Numeric(NumericType::NativeField), Type::Numeric(NumericType::NativeField)) => {
+                // Field -> Field: use src value
+                SimplifiedTo(value)
+            }
+            (
+                Type::Numeric(NumericType::Unsigned { .. }),
+                Type::Numeric(NumericType::NativeField),
+            ) => {
+                // Unsigned -> Field: redefine same constant as Field
+                SimplifiedTo(dfg.make_constant(constant, dst_typ.clone()))
+            }
+            (
+                Type::Numeric(NumericType::NativeField | NumericType::Unsigned { .. }),
+                Type::Numeric(NumericType::Unsigned { bit_size }),
+            ) => {
+                // Field/Unsigned -> unsigned: truncate
+                let integer_modulus = BigUint::from(2u128).pow(*bit_size);
+                let constant: BigUint = BigUint::from_bytes_be(&constant.to_be_bytes());
+                let truncated = constant % integer_modulus;
+                let truncated = FieldElement::from_be_bytes_reduce(&truncated.to_bytes_be());
+                SimplifiedTo(dfg.make_constant(truncated, dst_typ.clone()))
+            }
+            _ => None,
+        }
+    } else if let Some(value) = (*dst_typ == dfg.type_of_value(value)).then_some(value) {
+        SimplifiedTo(value)
```

Comment:
```suggestion
    } else if *dst_typ == dfg.type_of_value(value) {
        SimplifiedTo(value)
```
I think this was a remnant from when simplify returned an Option directly